<?xml version="1.0"?>
<!--
Copyright (c) 2020, STEREOLABS.

All rights reserved.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-->
<launch>
    <arg name="publish_urdf"         default="true" />
    <arg name="svo_file"             default="" /> <!-- <arg name="svo_file" default="path/to/svo/file.svo"> -->
    <arg name="stream"               default="" /> <!-- <arg name="stream" default="<ip_address>:<port>"> -->

    <arg name="node_name_1"          default="zed_node_left" />
    <arg name="camera_model_1"       default="zed2" />

    <arg name="camera_name_1"        default="zed_left" />

    <arg name="base_frame"           default="camera_center" /> 

    <arg name="cam_pos_x_1"             default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_y_1"             default="0.2675" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_z_1"             default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_roll_1"              default="-1.57079633" /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_pitch_1"             default="0.0" /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_yaw_1"               default="0.0" /> <!-- Orientation respect to base frame (i.e. "base_link) -->

    <group ns="$(arg camera_name_1)">
        <include file="$(find zed_wrapper)/launch/include/zed_camera.launch.xml">
            <arg name="camera_name"       value="$(arg camera_name_1)" />
            <arg name="svo_file"          value="$(arg svo_file)" />
            <arg name="stream"            value="$(arg stream)" />
            <arg name="node_name"         value="$(arg node_name_1)" />
            <arg name="camera_model"      value="$(arg camera_model_1)" />
            <arg name="base_frame"          value="$(arg base_frame)" />
            <arg name="publish_urdf"      value="$(arg publish_urdf)" />
	    <!--<arg name="publish_tf"        value="true" />-->
            <arg name="cam_pos_x"         value="$(arg cam_pos_x_1)" />
            <arg name="cam_pos_y"         value="$(arg cam_pos_y_1)" />
            <arg name="cam_pos_z"         value="$(arg cam_pos_z_1)" />
            <arg name="cam_roll"          value="$(arg cam_roll_1)" />
            <arg name="cam_pitch"         value="$(arg cam_pitch_1)" />
            <arg name="cam_yaw"           value="$(arg cam_yaw_1)" />
        </include>
    </group>
</launch>
